webapp/channels/src/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx

```diff
@@ -21,34 +21,32 @@ export interface Props extends PropsFromRedux {
     currentTeamName: string;
 }
 
-export default class SidebarBaseChannel extends React.PureComponent<Props> {
-    handleLeavePublicChannel = (callback: () => void) => {
-        this.props.actions.leaveChannel(this.props.channel.id);
+const SidebarBaseChannel = ({channel, currentTeamName, actions}: Props) => {
+    const intl = useIntl();
+
+    const handleLeavePublicChannel = (callback: () => void) => {
```

Comment:
Even if assigned by `getChannelLeaveHandler`, this and `handleLeavePrivateChannel` are used as props, so we should wrap them in `useCallback` to avoid sending one new function on every render.